src/app/server/Server.cpp

```diff
@@ -170,6 +170,8 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
     mGroupsProvider = initParams.groupDataProvider;
     SetGroupDataProvider(mGroupsProvider);
 
+    mReportScheduler = initParams.reportScheduler;
```

Comment:
doesn't this imply that the application can change the reportScheduler used?

Reply:
If we have a configuration define in `ChipConfig.h`, why would we have the reportScheduler in the initParams?
Unless we want to leave application create their own version of the handler, i don't the point of having it in the initParams.

i think we only one of the two options, not both?

Reply:
Sorry I closed without refreshing and I only saw the 2 first comments. That being said, I agree, since we don't do it now, let's just use the ifdef and we can inject it later if we judge it necessary.